fv3core/stencils/delnflux.py

```diff
@@ -1,94 +1,404 @@
     with computation(PARALLEL), interval(...):
-        fy2[0, 0, 0] = del6_u * (q[0, -1, 0] - q)
-        fy2[0, 0, 0] = fy2 * -1 if order > 1 else fy2
+        fy[0, 0, 0] = del6_u * (q[0, -1, 0] - q)
+        fy[0, 0, 0] = fy * -1 if order > 1 else fy
+
+
+@gtscript.function
+def fx_calculation(q: FloatField, del6_v: FloatField, order: int):
+    fx = del6_v * (q[-1, 0, 0] - q)
+    fx = -1.0 * fx if order > 1 else fx
+    return fx
+
+
+@gtscript.function
+def fy_calculation(q: FloatField, del6_u: FloatField, order: int):
+    fy = del6_u * (q[0, -1, 0] - q)
+    fy = fy * -1 if order > 1 else fy
+    return fy
 
 
 # WARNING: untested
 @gtstencil()
-def fx2_firstorder_use_sg(q: sd, sin_sg1: sd, sin_sg3: sd, dy: sd, rdxc: sd, fx2: sd):
+def fx_firstorder_use_sg(
+    q: FloatField,
+    sin_sg1: FloatField,
+    sin_sg3: FloatField,
+    dy: FloatField,
+    rdxc: FloatField,
+    fx: FloatField,
+):
     with computation(PARALLEL), interval(...):
-        fx2[0, 0, 0] = (
+        fx[0, 0, 0] = (
             0.5 * (sin_sg3[-1, 0, 0] + sin_sg1) * dy * (q[-1, 0, 0] - q) * rdxc
         )
 
 
 # WARNING: untested
 @gtstencil()
-def fy2_firstorder_use_sg(q: sd, sin_sg2: sd, sin_sg4: sd, dx: sd, rdyc: sd, fy2: sd):
+def fy_firstorder_use_sg(
+    q: FloatField,
+    sin_sg2: FloatField,
+    sin_sg4: FloatField,
+    dx: FloatField,
+    rdyc: FloatField,
+    fy: FloatField,
+):
     with computation(PARALLEL), interval(...):
-        fy2[0, 0, 0] = (
+        fy[0, 0, 0] = (
             0.5 * (sin_sg4[0, -1, 0] + sin_sg2) * dx * (q[0, -1, 0] - q) * rdyc
         )
 
 
 @gtstencil()
-def d2_highorder(fx2: sd, fy2: sd, rarea: sd, d2: sd):
+def d2_highorder(fx: FloatField, fy: FloatField, rarea: FloatField, d2: FloatField):
     with computation(PARALLEL), interval(...):
-        d2[0, 0, 0] = (fx2 - fx2[1, 0, 0] + fy2 - fy2[0, 1, 0]) * rarea
+        d2[0, 0, 0] = (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea
 
 
-@gtstencil()
-def d2_damp(q: sd, d2: sd, damp: float):
-    with computation(PARALLEL), interval(...):
-        d2[0, 0, 0] = damp * q
+@gtscript.function
+def d2_high_order(fx: FloatField, fy: FloatField, rarea: FloatField):
+    d2 = (fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) * rarea
+    return d2
 
 
 @gtstencil()
-def add_diffusive(fx: sd, fx2: sd, fy: sd, fy2: sd):
+def d2_damp(q: FloatField, d2: FloatField, damp: float):
     with computation(PARALLEL), interval(...):
-        fx[0, 0, 0] = fx + fx2
-        fy[0, 0, 0] = fy + fy2
+        d2[0, 0, 0] = damp * q
 
 
 @gtstencil()
-def add_diffusive_component(fx: sd, fx2: sd):
+def add_diffusive_component(
+    fx: FloatField, fx2: FloatField, fy: FloatField, fy2: FloatField
+):
+    from __externals__ import local_ie, local_je
+
     with computation(PARALLEL), interval(...):
-        fx[0, 0, 0] = fx + fx2
+        with horizontal(region[:, : local_je + 1]):
+            fx[0, 0, 0] = fx + fx2
+        with horizontal(region[: local_ie + 1, :]):
+            fy[0, 0, 0] = fy + fy2
```

Comment:
I assume these local regions are necessary to validate? Curious, if you don't include them, I am guessing delnflux and d_sw will fail, but will dyncore? Would be helpful to mark whether these regions are just needed for local validation, or if they are important all the way up the chain. Usually would look for the things using fx and fy to see if gt4py could extend the domain for us here, but looks like it's pretty far away from that, so we probably need these for now.